src/services/findAllReferences.ts

```diff
@@ -2004,20 +2023,51 @@ namespace ts.FindAllReferences {
 
         function getReferencesForStringLiteral(node: StringLiteralLike, sourceFiles: readonly SourceFile[], checker: TypeChecker, cancellationToken: CancellationToken): SymbolAndEntries[] {
             const type = getContextualTypeFromParentOrAncestorTypeNode(node, checker);
+            const callExpr = findAncestor(node, isCallExpression);
             const references = flatMap(sourceFiles, sourceFile => {
                 cancellationToken.throwIfCancellationRequested();
                 return mapDefined(getPossibleSymbolReferenceNodes(sourceFile, node.text), ref => {
-                    if (isStringLiteralLike(ref) && ref.text === node.text) {
-                        if (type) {
-                            const refType = getContextualTypeFromParentOrAncestorTypeNode(ref, checker);
-                            if (type !== checker.getStringType() && type === refType) {
-                                return nodeEntry(ref, EntryKind.StringLiteral);
-                            }
-                        }
-                        else {
+                    // Special case: Every string literal refers at least to itself.
+                    if (node === ref) return nodeEntry(ref, EntryKind.StringLiteral);
+
+                    // When evaluating references for a string literal, guarantee that the string literal
+                    // node is either global or that any reference shares the same scope.  Global literals
+                    // might all refer to each other, but if the string literal being considered is in a
+                    // call expression, any reference should also be from the same call expression.
+                    const refHasSameScope = !callExpr || callExpr === findAncestor(ref, isCallExpression);
+                    const refHasSameText = isStringLiteralLike(ref) || isPropertyNameLiteral(ref) && getTextOfIdentifierOrLiteral(ref) === node.text;
+
+                    if (type && isStringLiteralLike(ref) && refHasSameText) {
+                        const refType = getContextualTypeFromParentOrAncestorTypeNode(ref, checker);
+                        if ((refHasSameScope || !type.isStringLiteral() || type.isUnionOrIntersection()) && type === refType) {

```

Comment:
`!type.isStringLiteral()` is different from the old check `type !== checker.getStringType()`. Does that invert the set of refs that will be marked as references?